### ISSUE TITLE
[BUG] agent should ensure it can launch a new job before doing so

### ISSUE THREAD

**Issue by cjdcordeiro** [nuvlabox/agent#74]
**Describe the bug**
if a previous job got stuck, it will create a conflict when the agent tried to run it again

**To Reproduce**
example:

```
Processing the following jobs in pull-mode: ['job/9d63d181-d900-4ec4-8810-9337c6acbdfe', 'job/9eeea2e9-3a5e-48fd-b5e2-e61edd2945b5', 'job/8f1974a2-fac9-423b-9759-7dc300d7a1db', 'job/d28a2aa3-e49c-4aa7-b43e-ce9c3222b7c9', 'job/5d8805b2-0295-4ef2-b5dc-eb8ddcdb8897']
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/docker/api/client.py", line 268, in _raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: http+docker://localhost/v1.41/containers/f788157ba0b871ac0f972d9d343e702c4bf19a2bf37010d8905008770ef334d3/kill

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/nuvlabox/./app.py", line 191, in <module>
    job = Job(data_volume, job_id, infra.job_engine_lite_image)
  File "/opt/nuvlabox/agent/Job.py", line 32, in __init__
    self.do_nothing = self.check_job_is_running()
  File "/opt/nuvlabox/agent/Job.py", line 53, in check_job_is_running
    job_container.kill()
  File "/usr/local/lib/python3.9/site-packages/docker/models/containers.py", line 276, in kill
    return self.client.api.kill(self.id, signal=signal)
  File "/usr/local/lib/python3.9/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/docker/api/container.py", line 798, in kill
    self._raise_for_status(res)
  File "/usr/local/lib/python3.9/site-packages/docker/api/client.py", line 270, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/usr/local/lib/python3.9/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.APIError: 409 Client Error for http+docker://localhost/v1.41/containers/f788157ba0b871ac0f972d9d343e702c4bf19a2bf37010d8905008770ef334d3/kill: Conflict ("Cannot kill container: f788157ba0b871ac0f972d9d343e702c4bf19a2bf37010d8905008770ef334d3: Container f788157ba0b871ac0f972d9d343e702c4bf19a2bf37010d8905008770ef334d3 is not running")
```

this happens because `f788157ba0b8   nuvla/job-lite:2.14.2                         "/app/my_init -- /ap…"   About an hour ago   Created                                                      job-9d63d181-d900-4ec4-8810-9337c6acbdfe
`

**Comment by schaubl** [comment]
This seems to have been fixed in 1ed84a9b5bfd50ea498e331e9dd9b8150b11acaf

**Comment by cjdcordeiro** [comment]
indeed
